### ISSUE TITLE
Undocumented dependency of 'runtime'?

### ISSUE THREAD

**Issue by shehi** [pusher/pusher-js#549]
This line https://github.com/pusher/pusher-js/blob/master/src/core/auth/pusher_authorizer.ts#L4 shows a dependency on `runtime` package. Yet that package isn't listed within `package.json`.

Not that I recommend using that package, because as of now, it has a dependency on outdated version of `lodash` which has critical security issue. I don't expect `runtime` to be updated anytime soon, as 5 yrs has passed since the last time it was updated.

So, is it possible to replace `runtime` with something else? My problem is, I can't compile my code:

```
Failed to compile.

./node_modules/pusher-js/src/core/auth/pusher_authorizer.ts:4:21
Type error: Cannot find module 'runtime' or its corresponding type declarations.

  2 | import Channel from '../channels/channel';
  3 | import Factory from '../utils/factory';
> 4 | import Runtime from 'runtime';
    |                     ^
  5 | import { AuthTransports } from './auth_transports';
  6 | import {
  7 |   AuthOptions,
``` 

and installing `runtime` is problematic, since it doesn't pass audit checks.

**Comment by gileswells** [comment]
Unless I'm reading this wrong, `pusher-js` doesn't use the `runtime` dependency at all. We use the term `runtime` but it's as part of a webkit module resolution to allow us to make this singular project runtime agnostic between web, ReactNative, Node and worker contexts.

You can see more here: https://github.com/pusher/pusher-js/blob/master/README.md#core-vs-platform-specific-code

As for the critical result when running `npm audit` I've proposed a PR to the engineering team to review for potential merge.

**Comment by shehi** [comment]
I get your point regarding abstraction of "runtime". But as far as I know, the syntax of `import Something from 'something`` literally means a lib called `something` needs to be installed. If you want to import something from within `src/` folder, `something` needs to become a path, e.g. `./something` or `/something`. I am sure you already these things. So I fail to understand how you expect transpiler not to expect a library at that point in code.

As to my project, it is a NextJs project. I use pusher-js through laravel-echo. During development, while `npm run dev` is up, I get no errors at all (while I don't have `runtime` lib installed). But the moment I run `npm run build` on a NextJs app to get a production build, the abovementioned error occurs.

All in all, I won't be able to use pusher-js if this error remains there. :(

**Comment by shehi** [comment]
So guess there is nothing you can help me with @gileswells

**Comment by benw-pusher** [comment]
We are still looking into this and will update shortly.

**Comment by benw-pusher** [comment]
Fixed by #554

**Comment by amad** [comment]
@shehi We just upgraded all the old dependencies and resolved audit errors. 

As I understood you had an issue with `runtime`. As it was explained above, we don't use `runtime` npm package. We have an internal module called `runtime` for each context (e.g. node, react, web) and we use Webpack module resolution to map it to the right module on each context. Example: https://github.com/pusher/pusher-js/blob/master/webpack/config.node.js#L19

If you still have this problem, please open a new issue and explain how we can reproduce it and include details about your environment (node, typescript, npm, webpack versions) and steps needed to reproduce this error.
